ext/curl/interface.c

```diff
@@ -801,14 +801,16 @@ static int curl_ssh_hostkeyfunction(void *clientp, int keytype, const char *key,
 		php_error_docref(NULL, E_WARNING, "Cannot call the CURLOPT_SSH_HOSTKEYFUNCTION");
 	} else if (!Z_ISUNDEF(retval)) {
 		_php_curl_verify_handlers(ch, /* reporterror */ true);
-		zend_long retval_long = zval_get_long(&retval);
-		if (retval_long == CURLKHMATCH_OK || retval_long == CURLKHMATCH_MISMATCH) {
-			rval = retval_long;
+		if (Z_TYPE(retval) == IS_LONG) {
+			zend_long retval_long = zval_get_long(&retval);
```

Comment:
```suggestion
			zend_long retval_long = Z_LVAL_P(retval);
```

Reply:
This PR truly seems like a never ending story :p, I'll fix it, thx.

Reply:
Don't worry :) That's usually the case for non-trivial changes